### PR TITLE
fix(ingest/dremio): update dremio sql query to retrieve queried datasets in sql jobs 

### PR DESCRIPTION
**Pull request by acrylJonny**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by mayurinehate — APPROVED** [review]

File: metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py [hunk 1]
```diff
@@ -251,7 +251,7 @@ class DremioSQLQueries:
         SELECT
             *
         FROM
-            SYS.PROJECT.HISTORY.JOBS
+            sys.project.history.jobs
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py:254]: Is this case change necessary ? Is it also required for other sql queries from this file ?

Comment (acrylJonny) [reply metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py:254]: @dremio-brock has been doing some testing and found that this query specifically was giving a null pointer exception with Dremio unless this was changed to lowercase. This is a workaround to a Dremio Cloud bug.

File: metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py [hunk 2]
```diff
@@ -242,7 +242,7 @@ class DremioSQLQueries:
         SYS.JOBS_RECENT
     WHERE
         STATUS = 'COMPLETED'
-        AND LENGTH(queried_datasets)>0
+        AND ARRAY_SIZE(queried_datasets)>0
```
Comment (dremio-brock) [inline metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py:245]: On Software, the datatype is a varchar, LENGTH should be used.

Comment (acrylJonny) [reply metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py:245]: https://github.com/datahub-project/datahub/pull/11817

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py:245]: Dremio Software docs suggest queried_datasets is an array of varchar  `[varchar]`-> https://docs.dremio.com/current/reference/sql/system-tables/jobs_recent/#fields

Is the type in the doc incorrect ?

Comment (acrylJonny) [reply metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py:245]: For Dremio SW this is a varchar - the docs are incorrect though as it states an array of strings, i.e. [varchar] vs varchar. In SW this is coming through as a varchar (although it is a string that is an array representation), whereas on Cloud this correctly comes through as an array of strings. there is a PR open to fix this [here](https://github.com/datahub-project/datahub/pull/11817)

Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_sql_queries.py:245]: Thanks for confirming. Can you please add this inconsistency of queried_datasets  type as NOTE comment on fix PR ?

**Pull request merged**